querybook/webapp/components/DataDocSchedule/DataDocScheduleForm.tsx

```diff
@@ -284,6 +296,161 @@ export const DataDocScheduleForm: React.FunctionComponent<
+        `${notificationFormName}.config.to_all`
+    );
+
+    return (
+        <div className="cell-export-field mb24 flex-row">
+            <div className="flex1 mr16">
+                <div className="horizontal-space-between">
+                    <WrappedFormField
+                        label="Notify With"
+                        name={`${notificationFormName}.with`}
+                        type="react-select"
+                        options={notifierOptions}
+                        withDeselect
+                    />
+
+                    <WrappedFormField
+                        label="Notify On"
+                        name={`${notificationFormName}.on`}
+                        type="react-select"
+                        isDisabled={!notificationRow.with}
+                        options={notifyOnOptions}
+                    />
+                </div>
+
+                <FormField
+                    label="Notify To"
+                    help={getHelp(notificationRow.with)}
+                    error={
+                        notifyToAllMeta.touched ? notifyToAllMeta.error : null
+                    }
+                >
+                    <MultiCreatableUserSelect
+                        value={
+                            notifyToAllMeta.value ??
+                            notifyToAllMeta.initialValue
+                        }
+                        onChange={notifyToAllHelpers.setValue}
+                        selectProps={{
+                            isClearable: true,
+                            placeholder: getHelp(notificationRow.with),
+                            onBlur: () => notifyToAllHelpers.setTouched(true),
+                        }}
+                    />
+                </FormField>
+            </div>
+            <div>
+                <IconButton icon="X" onClick={handleRemoveNotifier} />
+            </div>
+        </div>
+    );
+};
+
+const ScheduleNotifactionsForm: React.FC<{
+    notifiers: INotifier[];
+}> = ({ notifiers }) => {
+    const name = 'kwargs.notifications';
+    const { values } = useFormikContext<IScheduleFormValues>();
+
+    const notificationValues = values.kwargs.notifications ?? [];
+
+    const notifierOptions = useMemo(
+        () => notifiers.map((notifier) => notifier.name),
+        [notifiers]
+    );
+
+    const notifyOnOptions = useMemo(
+        () =>
+            getEnumEntries(NotifyOn).map(([key, value]) => ({
+                value,
+                label: key,
+            })),
+        []
+    );
+
+    const getNotifierHelp = useCallback(
+        (notifierName: string) =>
+            notifiers.find((n) => n.name === notifierName)?.help ||
+            'Add comma(,) separated recepients here',
+        [notifiers]
+    );
+
+    const handleNewNotification = useCallback(
+        (arrayHelpers) => {
+            arrayHelpers.push({
+                with: notifierOptions[0] ?? null,
```

Comment:
`?? null` is pretty much noop unless we need to differentiate between null and undefined